src/ring_theory/derivation.lean

```diff
@@ -0,0 +1,240 @@
+/-
+Copyright © 2020 Nicolò Cavalleri. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Nicolò Cavalleri.
+-/
+
+import algebra.lie_algebra
+import tactic
+
+/-!
+# Derivations
+
+This file defines derivation. A derivation from the `R`-agebra `A` to the `A`-module `M` is an `R`
+linear map that satisfy the Leibniz rule.
+
+## Notation
+
+The notation `⁅D1, D2⁆` is used for the commutator of two derivations.
+
+TODO: this file is just a stub to go on with some PRs in the geometry section. It only
+implements the definition of derivations in commutative algebra. This will soon change: as soon
+as bimodules will be there in mathlib I will change this file to take into account the
+non-commutative case. Any development on the theory of derivations is discouraged until the
+definitive definition of derivation will be implemented.
+-/
+
+open algebra ring_hom
```

Comment:
It looks to me like some of the definitions here overlap with the restriction of scalars formalism already [here](https://github.com/leanprover-community/mathlib/blob/b1e56a21337e08dab0473d66aac8d53ef278aa56/src/ring_theory/algebra.lean#L1123).

Do you think it would be possible to build on top of that instead?

Reply:
Ops yeah you are right. Fixed.